api/nodes/serializers.py

```diff
@@ -46,7 +63,14 @@ def get_absolute_url(self, obj):
         return obj.absolute_url
 
     def get_node_count(self, obj):
-        return len(obj.nodes)
+        request = self.context['request']
+        user = request.user
+        if user.is_anonymous():
+            auth = Auth(None)
+        else:
+            auth = Auth(user)
+        nodes = [node for node in obj.nodes if node.can_view(auth) and node.primary]
```

Comment:
Those are more for optional filters above and beyond the default, while this is more of a default implementation. It really should be restricted on the model, but it's not and won't be for a while. It might be that we can move the count to the view where we're already doing this logic in other circumstances, but we'd lose some of the advantages of the ODM queries if we did that, I think. 

That being said, maybe it could be put onto a default filter? I'll give it some more thought. Maybe so.